### PR TITLE
Support leaseweb culprits ipv6

### PR DESCRIPTION
**Pull request by furkansahin**
## Remove the prefix to be shorter than 68 bits condition
We were expecting the prefix to be shorter than 68 bits but the latest
leaseweb hosts came with ipv6 blocks like:
2607:f5b7:1:64:1::_112/64
2607:f5b7:1:64:2::_112/64
2607:f5b7:1:64:3::_112/64

As you can see, they give /64 but the address space is not actually as
big as /64. Because, the :1, :2, :3 at the end of the prefix is actually
to 80th bit. We must see them like the following for it to make sense;
2607:f5b7:1:64:0001::/64
2607:f5b7:1:64:0002::/64

Here, the differentiating bit is in the residue part of a /64 network.
So, if we use them as they are, we need to actually map them to
2607:f5b7:1:64::/64 which would not work because then we would have
multiple hosts with the same prefix and we cannot assign ipv6 addresses
to the VMs correctly.

To solve this issue, we assign the prefixes correctly using /80. So, the
hosts become
2607:f5b7:1:64:0001::/80
2607:f5b7:1:64:0002::/80

This requires modification in learn_network prog because it was
previously expecting at least /68. We remove the condition.

The main idea of /68 was that so that we can delegeta prefixes at the
size of /80 which doesn't really work because dnsmasq cannot delegate
the prefixes anyway. So, in any case, we simply give /128 to anyone. So,
I remove that condition.

## Update vm_host to support allocation of longer prefixes to VMs
We always tried to allocate /79 to the VM to be used as 2 /80s.This
doesn't work with the new leaseweb hosts because we already get a /80
from them for the host. Therefore, we need to be able to allocate much
smaller. Keeping the entropy the same (2 bytes), we change the ipv6
allocation logic. This way, hetzner hosts will continue allocation /79
while leaseweb allocates /95.

## Use proxy ndp for both clover and guest ephemeral
With the recent change to use dnsmasq as the first level dns server, we
get out to the internet with the clover_ephemeral address. The existing
ndp proxy logic was only proxying the guest ephemeral which makes the VM
IPv6 stack working as expected. However, we realized that the dns
queries through dnsmasq was failing. Further investigation showed that
the dnsmasq was able to send packets out but no response was getting
back in. I realized the source address is the first half of the /95
prefix for dnsmasq. Therefore, we need to enable ndp_proxy for that
prefix, too. This way, our response packets started to flow back in.

### CODE REVIEW

File: spec/model/vm_host_spec.rb [hunk 1]
```diff
@@ -64,10 +64,10 @@
     expect(vh.ip6_random_vm_network.to_s).not_to eq(vh.ip6_reserved_network)
   end
 
-  it "can generate ipv6 for hosts with smaller than /64 prefix with single byte" do
+  it "can generate ipv6 for hosts with smaller than /64 prefix with two bytes" do
     vh.net6 = NetAddr.parse_net("2a01:4f9:2b:35a::/68")
-    expect(SecureRandom).to receive(:random_number).with(2...256).and_return(5)
-    expect(vh.ip6_random_vm_network.to_s).to eq("2a01:4f9:2b:35a:4::/79")
+    expect(SecureRandom).to receive(:random_number).with(2...2**16).and_return(5)
+    expect(vh.ip6_random_vm_network.to_s).to eq("2a01:4f9:2b:35a:0:4000:0:0/83")
```
Comment (fdr) [inline spec/model/vm_host_spec.rb:70]: I would expand this test to do some boundary conditions: the lowest value, the highest value. I guess a middle value for good measure, and freeze the result of that bit-bashing into the test.

File: model/vm_host.rb [hunk 2]
```diff
@@ -85,8 +86,7 @@ def ip6_random_vm_network(prefix = 79)
     # The upper range is updated later to increase the entropy if the host has
```
Comment (fdr) [inline model/vm_host.rb:86]: Are these descriptions still legit? Seems like the algorithm/design theory has changed quite a bit?

File: spec/model/vm_host_spec.rb (hunk 1 above)
Comment (fdr) [reply spec/model/vm_host_spec.rb:70]: I might also consider an all-zero address as the test fixture, to make sure the boundaries of your shifting are right, since a bitwise-or will make ambiguous whether a set bit is from the test fixture (is it set from the get-go, or did the `|` set it?)

File: model/vm_host.rb (hunk 2 above)
Comment (furkansahin) [reply model/vm_host.rb]: Updated them

File: spec/model/vm_host_spec.rb (hunk 1 above)
Comment (furkansahin) [reply spec/model/vm_host_spec.rb:70]: Added new tests

**Review by fdr — APPROVED** [review]

**Comment by fdr** [comment]
@pykello you are a skilled bit-basher, could you double-check this?
